bentoml/deployment/sagemaker/__init__.py

```diff
@@ -377,21 +377,21 @@ def _create_sagemaker_model(
             "ContainerHostname": sagemaker_model_name,
             "Image": ecr_image_path,
             "Environment": {
-                "API_NAME": bento_service_api_name,
+                "API_NAME": spec.api_name,
                 "BENTO_SERVER_TIMEOUT": config().get('apiserver', 'default_timeout'),
             },
         },
         "ExecutionRoleArn": execution_role_arn,
     }
-    default_worker_count = config().getint(
-        'apiserver', 'default_gunicorn_workers_count'
-    )
-    if default_worker_count > 0:
+
+    # Will set envvar, if user defined gunicorn worker is greater than 0.  EnvVar needs
+    # to be string instead of the int.
+    default_num_of_gunicorn_worker_count = spec.num_of_gunicorn_workers_per_instance
```

Comment:
IIUC, num_of_gunicorn_workers_per_instance is optional right?

shouldn't this be:
```
if spec.num_of_gunicorn_workers_per_instance:
   sagemaker_model_info['PrimaryContainer']['Environment']['GUNICORN_WORKER_COUNT'] = ....
```

Reply:
I suppose this code assumes we have something like a default value of "-1" for num_of_gunicorn_workers_per_instance,  but I don't see where we set that default value for this proto field

Reply:
yeah, this is assuming the user doesn't enter any non-positive value.  Since we already check the value during validation, i probably should update it to what you type out